rita/src/rita_common/peer_listener/mod.rs

```diff
@@ -0,0 +1,397 @@
+        }
+        Ok(res.clone())
+    }
+}
+
+#[derive(Debug)]
+pub struct GetPeers();
+impl Message for GetPeers {
+    type Result = Result<Vec<Peer>, Error>;
+}
+
+impl Handler<GetPeers> for PeerListener {
+    type Result = Result<Vec<Peer>, Error>;
+
+    fn handle(&mut self, _: GetPeers, _: &mut Context<Self>) -> Self::Result {
+        Ok(self.peers.clone())
+    }
+}
+
+#[derive(Debug)]
+pub struct ListenInterface {
+    ifname: String,
+    ifidx: u32,
+    multicast_socketaddr: SocketAddrV6,
+    multicast_socket: UdpSocket,
+    linklocal_socket: UdpSocket,
+    linklocal_ip: Ipv6Addr,
+}
+
+impl ListenInterface {
+    pub fn new(ifname: &str) -> Result<ListenInterface, Error> {
+        let port = SETTING.get_network().rita_hello_port;
+        let disc_ip = SETTING.get_network().discovery_ip;
+        trace!("Binding to {:?} for ListenInterface", ifname);
+        // Lookup interface link local ip
+        let link_ip = match KI.get_link_local_device_ip(&ifname) {
+            Ok(ip) => ip,
+            Err(e) => {
+                return Err(e);
+            }
+        };
+        // Lookup interface index
+
+        let iface_index = match KI.get_iface_index(&ifname) {
+            Ok(idx) => idx,
+            Err(_) => 0,
+        };
+        // Bond to multicast discovery address on each listen port
+        let multicast_socketaddr = SocketAddrV6::new(disc_ip, port.into(), 0, iface_index);
+        let multicast_socket = UdpSocket::bind(multicast_socketaddr)
+            .expect("Failed to bind to peer discovery address!");
+        let res = multicast_socket.join_multicast_v6(&disc_ip, iface_index);
+        trace!("ListenInterface init set multicast v6 with {:?}", res);
+        let res = multicast_socket.set_nonblocking(true);
+        trace!(
+            "ListenInterface multicast init set nonblocking with {:?}",
+            res
+        );
+
+        let linklocal_socketaddr = SocketAddrV6::new(link_ip, port.into(), 0, iface_index);
+        let linklocal_socket = UdpSocket::bind(linklocal_socketaddr).expect(&format!(
+            "ListenInterface Failed to bind to link local address {:?} on {:?} with iface_index {:?} ",
+            link_ip, ifname, iface_index
+        ));
+        let res = linklocal_socket.set_nonblocking(true);
+        trace!("ListenInterface init set nonblocking with {:?}", res);
+
+        let res = linklocal_socket.join_multicast_v6(&disc_ip, iface_index);
+        trace!("ListenInterface Set link local multicast v6 with {:?}", res);
+
+        Ok(ListenInterface {
+            ifname: ifname.to_string(),
+            ifidx: iface_index,
+            multicast_socket: multicast_socket,
+            linklocal_socket: linklocal_socket,
+            multicast_socketaddr: multicast_socketaddr,
+            linklocal_ip: link_ip,
+        })
+    }
+}
+
+fn encode_im_here(addr: Ipv6Addr) -> Vec<u8> {
+    let mut buf = Vec::new();
+    buf.put_u32_be(MSG_IM_HERE);
+    buf.put_u16_be(22);
```

Comment:
We'll never exceed MTU so usually it would be below ~1500bytes at all times. If anything I'd just shrink magic key to 16 bits to save 2 bytes. In theory magic key could be even `u8` as we won't exceed 255 distinct message types. But at least for now it could be like this as well.

Reply:
I made the magic long to make attempts to parse other protocols messages less probable, but even being on the right port and ip is already vanishingly unlikely, I'll go for u8